### DISCUSSION TITLE
I can't intercept a request to Next.js API Route

### DISCUSSION THREAD

**Discussion by peelar** [cypress-io/cypress#15726] [Questions and Help]
Hello everyone!

Sandbox here: https://codesandbox.io/s/competent-gould-i6od8?file=/pages/index.js

A big fan of Cypress here, somehow this tool made me excited about testing for the first time ever. However, I have an issue that's been blocking me for some time. I have a Next.js application with one test endpoint `api/ping`. It looks like this:

```
// pages/api/ping.js
export default (req, res) => {
  setTimeout(() => {
    res.status(200).json({})
  }, 3000);
}
```

In my `Form` component, I am sending a request to it like this:

```
// src/components/Form.js
...
const sendForm = () =>
  fetch('/api/ping', {
    method: 'POST',
  })
...
```

Everything works just fine in the actual application, the request runs successfully. However, I can't intercept it any way I try. My current test looks like this:

```
// cypress/integration/form.js
    ...
    cy.intercept('POST', '/api/ping', {
      statusCode: 200,
      body: {},
    })
  })
})
```

I want to intercept the request so that the UI doesn't have to wait 3 seconds for that `setTimeout` to end. However, the requests don't get intercepted at all 😢. Does anyone have any idea what is the cause here?

**Comment by mathiasgheno** [comment]
Isn't because you're trying to make the `cy.intercept` after the submit operation?

**Reply by mathiasgheno** [reply]
Yep, is that. I tried to do it in the first line of your test:

```javascript
cy.intercept('POST', '/api/ping', {
      statusCode: 200,
      body: {
        hello: true,
      },
 })
```

![Screenshot from 2021-04-01 18-19-15](https://user-images.githubusercontent.com/14915619/113355408-1ae9e880-9317-11eb-980e-480555985d2e.png)

**Reply by mathiasgheno** [reply]
Try to use yours `cy.intercept` in the `beforeEach` and it should solve. :D

**Reply by peelar** [reply]
🤦 Thank God it's Friday already. Thanks man, I owe you a 🍺.
